crates/aptos-dkg/src/range_proof.rs

```diff
@@ -465,23 +495,37 @@ fn byte_to_bits_le(val: u8) -> Vec<bool> {
 
 /// Compute alpha, beta.
 fn fiat_shamir_challenges(
-    _com: &Commitment,
+    com: &Commitment,
     c: &[G1Projective],
     c_hat: &[G2Projective],
-    with_alphas: bool,
+    with_alphas: bool, // atm this has to be true for both prover and verifier, because FS transcripts have to be consistent
+    fs_transcript: &mut merlin::Transcript,
 ) -> (Vec<Scalar>, Vec<Scalar>) {
     assert_eq!(c.len(), c_hat.len());
-    //TODO: real fiat-shamir.
-    let mut rng = StdRng::from_seed([0; 32]);
-    let beta_vals = (0..c.len())
-        .map(|_| random_128bit_scalar(&mut rng))
-        .collect();
+
+    <merlin::Transcript as fiat_shamir::RangeProof<Scalar>>::append_range_proof_sep(fs_transcript);
+
+    // Add the first prover message to the transcript
```

Comment:
You need to additionally add:
1. VK
2. The full public statement: $\ell$ and $n$; you've only added $C$

I believe you added the first message, i.e., the $C_j$'s? But why are you not adding the $\tilde{C}_j$'s? (See [proving algorithm here](https://alinush.github.io/dekart#widetildemathsfdekartmathsffftmathsfprovemathcalfscdotmathsfprk-c-ell-z_0ldotsz_n-1-rrightarrow-pi).)

Reply:
The honest ones. AFAICT, they need to be added.